lib/internal/readline/promises.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const {
+  NumberIsNaN,
+  Promise,
+  PromiseReject,
+  PromiseResolve,
+} = primordials;
+
+const {
+  codes: {
+    ERR_INVALID_ARG_VALUE,
+    ERR_INVALID_CURSOR_POS,
+  },
+} = require('internal/errors');
+
+const {
+  CSI,
+} = require('internal/readline/utils');
+
+const {
+  kClearToLineBeginning,
+  kClearToLineEnd,
+  kClearLine,
+  kClearScreenDown,
+} = CSI;
+
+
+/**
+ * Moves the cursor to the x and y coordinate on the given stream.
+ */
+function cursorTo(stream, x, y = undefined) {
```

Comment:
The difference in usage is quite different from the callback API.
this can reject with an error, while the callback API could be used in a "fire and forget" effort, this one cannot be: it has to be awaited.

I think we should remove all the PromiseReject() for the validation and just make it throw.

PTAL: https://github.com/nodejs/node/blob/ea0154814a6151739129b1f13674d96135a6f1e8/lib/readline.js#L1320-L1346.

We are also losing the return value of the stream: it signals if we could write more and/or if we need to wait for the `'drain'` event. I think we should design this API differently.

Reply:
Is it really a problem if both API has different behaviour? I suppose users who need the `drain` info or the sync throwing should continue to use the callback-API, there's no plan to retire those I believe.

> I think we should remove all the PromiseReject() for the validation and just make it throw.

I don't know, some people may prefer a rejected promise, otherwise they should use the callback API.